frontend/packages/dev-console/integration-tests/features/topology/topology-chart-area-visual.feature

```diff
@@ -172,3 +172,86 @@ Feature: Topology chart area
+              And user can see General section with Application name and Name under it
+              And user can see Resources and Advanced options sections
+
+
+        @regression, @manual
+        Scenario: Drag and drop jar file in topology chart view
+            Given user has a jar file named "sample_yaml_upload.yaml"
+              And user is at the Topology page
+             When user drags and drop jar file on topology
+              And user gives Application name as "sample-upload-app" and workload Name as "sample-yaml-upload"
+              And user clicks on Create in Upload JAR file form
+             Then user is redirected to topology
+              And user can see a toast notification of JAR file uploading with link to build logs
+              And user can see deployment "sample-yaml-upload" in application "sample-upload-app" is created in topology
+
+
+        @regression @manual
+        Scenario: Add to Project to upload JAR file in topology
+            Given user is at the Topology page
+             When user right clicks on the empty chart area
+              And user clicks on Add to Project
+              And user clicks on Upload JAR file
+              And user clicks on Browse in JAR file section
+              And user selects file to upload
+              And user clicks Clear and reupload the file
+              And user gives Application name as "sample-upload-app" and workload Name as "sample-yaml-upload-1"
+              And user clicks on Create
+             Then user is redirected to topology
+              And user can see a toast notification of JAR file uploading with link to build logs
+              And user can see deployment "sample-yaml-upload-1" in application "sample-upload-app" is created in topology
+
+
+        @regression @manual
+        Scenario: Add to Project through drag and drop to upload JAR file in topology
+            Given user is at the Topology page
+             When user right clicks on the empty chart area
+              And user clicks on Add to Project
+              And user clicks on Upload JAR file
+              And user drag and drop the file in JAR file section
+              And user gives Application name as "sample-upload-app" and workload Name as "sample-yaml-upload-1"
+              And user clicks on Create
+             Then user is redirected to topology
+              And user can see a toast notification of JAR file uploading with link to build logs
+              And user can see deployment "sample-yaml-upload-1" in application "sample-upload-app" is created in topology
+
+
+        @regression, @manual
+        Scenario: Drag and drop Incompatible file in topology chart view
+            Given user has a incompatible file
+              And user is at the Topology chart view
+             When user drags and drop the file on topology
+             Then the curser will show the action is not available
```

Comment:
This is not possible. The user must first drop the file before we can validate it.

UX wants to display a toast warning message that the file is invalid.

Reply:
I've added about the toast warning message